src/r/recordFactories/ProjectFactory.ts

```diff
@@ -987,25 +997,30 @@ export class ProjectUtils {
             this.accentColorsDict[accentColor].push(ruleId) :
             this.accentColorsDict[accentColor] = [ruleId];
         }
-        
+
         // element and variable names entries
         const elements = sceneF.getAllDeepChildren(RT.element);
-
         const whenEvents = ruleF.getRecords(RT.when_event);
+
         for (const we of whenEvents) {
           // const whenEventF = new RecordFactory(we);
           const coId = we.props.co_id as number;
-          const element = elements.find(ele => ele.id === coId);
-          const variable = sceneF.getRecord(RT.variable, coId);
-          if (element) {
-            const eleName = element.name?.trim().toLowerCase();
+          // const element = elements.find(ele => ele.id === coId);
+          const element = sceneF.getAllDeepChildrenWithFilter(RT.element, (ele => ele.id === coId));
```

Comment:
Since you already have all the elements extracted at line#1002, you can re-use that variable and use the .find method to look for a valid element